### PR TITLE
feat: alerts REST client

### PR DESCRIPTION
**Pull request by kkedziak-splunk**
**Issue number:** ADDON-77073

### PR Type

**What kind of change does this PR introduce?**
* [x] Feature
* [ ] Bug Fix
* [ ] Refactoring (no functional or API changes)
* [ ] Documentation Update
* [ ] Maintenance (dependency updates, CI, etc.)

### Changes

REST client used to create, read, update and delete alerts.

### User experience

No changes to the existing code. New class added.

## Checklist

If an item doesn't apply to your changes, leave it unchecked.

* [x] I have performed a self-review of this change according to the [development guidelines](https://splunk.github.io/addonfactory-ucc-generator/contributing/#development-guidelines)
* [x] Tests have been added/modified to cover the changes [(testing doc)](https://splunk.github.io/addonfactory-ucc-generator/contributing/#build-and-test)
* [x] PR title and description follows the [contributing principles](https://splunk.github.io/addonfactory-ucc-generator/contributing/#pull-requests)

### CODE REVIEW

File: solnlib/alerts_rest_client.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+            name: Name of the alert.
+            search: Search query for the alert.
+            disabled: Whether the alert is disabled. Default is True.
+            description: Description of the alert.
+            alert_type: Type of the alert (see AlertType). If it equals to CUSTOM, Splunk executes a check in
+                alert_condition. Otherwise, alert_comparator and alert_threshold are used.
+            alert_condition: Condition for the alert.
+            alert_comparator: Comparator for the alert. Default is GREATER_THAN.
+            alert_threshold: Threshold for the alert. Default is 0.
+            time_window: Time window for the alert. Tuple of earliest and latest time. Default is ("-1h", "now").
```
Comment (hetangmodi-crest) [inline solnlib/alerts_rest_client.py:104]: typo: the default here and in function definition are different.

File: solnlib/alerts_rest_client.py [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+            context: Other configurations for Splunk rest client.
+        """
+        self.session_key = session_key
+        self.app = app
+
+        self._rest_client = rest_client.SplunkRestClient(
+            self.session_key, app=self.app, **context
+        )
+
+    def create_search_alert(
```
Comment (hetangmodi-crest) [inline solnlib/alerts_rest_client.py:75]: What would be the permission of the created alert - private, shared?

Comment (kkedziak-splunk) [reply solnlib/alerts_rest_client.py:75]: This is the value of `acl`:

```
{
  "app": "search",
  "can_change_perms": true,
  "can_list": true,
  "can_share_app": true,
  "can_share_global": true,
  "can_share_user": true,
  "can_write": true,
  "modifiable": true,
  "owner": "admin",
  "perms": null,
  "removable": true,
  "sharing": "user"
}
```

So the alert will be private to the user.

I didn't think about it. It should be shared through an app.

File: solnlib/alerts_rest_client.py (hunk 1 above)
Comment (kkedziak-splunk) [reply solnlib/alerts_rest_client.py:104]: I will correct that.

File: solnlib/alerts_rest_client.py (hunk 2 above)
Comment (kkedziak-splunk) [reply solnlib/alerts_rest_client.py:81]: Now by default alerts are created as shared.

**Review by hetangmodi-crest — APPROVED** [review]

**Pull request merged**